linkerd/app/outbound/src/zone.rs

```diff
@@ -0,0 +1,79 @@
+use crate::metrics::prom;
+use linkerd_app_core::{
+    metrics::OutboundZoneLocality, svc, transport::metrics::zone::TcpZoneMetrics,
+};
+use prometheus_client::{
+    encoding::{EncodeLabelSet, EncodeLabelValue},
+    registry::Unit,
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct TcpZoneMetricsParams {
+    transfer_cost: prom::Family<TcpZoneLabels, prom::Counter>,
+}
+
+impl TcpZoneMetricsParams {
+    pub fn register(registry: &mut prom::Registry) -> Self {
```

Comment:
It would be most idiomatic to move this into linkerd/transport-metrics/src/zone.rs and have it be generic over the label type -- then the metrics details can all be private to the module and the labeling context remains in the outbound app.

Reply:
Took a bit to unwind some orphan rule errors I was getting, but I agree this makes more sense.